samples/crud_extension/crud_handler.py

```diff
@@ -0,0 +1,104 @@
+            # Create a document
+            index_name = 'test-index'
+            document = request.content
+            response = self.client.index(
+                index=index_name,
+                body=document,
+                refresh=True
+            ) 
+            response_bytes = bytes(json.dumps(response).encode("utf-8"))
+            return ExtensionRestResponse(RestStatus.OK, response_bytes, ExtensionRestResponse.JSON_CONTENT_TYPE)
+        
+        elif request.method == RestMethod.GET:
+            # Search for documents
+            index_name = 'test-index'
+            q = request.params.get('q')
+            
+            query = {
+                'size': 5,
+                'query': {
+                    'multi_match': {
+                        'query': q,
+                        'fields': ['title^2', 'director']
+                    }
+                }
+            }
+            response = self.client.search(
+                body=query,
+                index=index_name
+            )
+            response_bytes = bytes(json.dumps(response).encode("utf-8"))
+            return ExtensionRestResponse(RestStatus.OK, response_bytes, ExtensionRestResponse.JSON_CONTENT_TYPE, consumed_params=['q'])
+        
+        elif request.method == RestMethod.PUT:
+            # Update a document
+            index_name = 'test-index'
+            id = request.params.get('id')
+            document = request.content
+            response = self.client.index(
+                index=index_name,
+                body=document,
+                id=id,
+                refresh=True
+            )
+            response_bytes = bytes(json.dumps(response), "utf-8")
+            return ExtensionRestResponse(RestStatus.OK, response_bytes, ExtensionRestResponse.JSON_CONTENT_TYPE, consumed_params=['id'])
+        
+        elif request.method == RestMethod.DELETE:
+            # Delete a document
+            index_name = 'test-index'
+            id = request.params.get('id')
+            response = self.client.delete(
+                index=index_name,
+                id=id
+            )
+            response_bytes = bytes(json.dumps(response), "utf-8")
+            logging.info(f"response: {response}")
+            return ExtensionRestResponse(RestStatus.OK, response_bytes, ExtensionRestResponse.JSON_CONTENT_TYPE, consumed_params=['id'])
+        
+        else:
+            return ExtensionRestResponse(RestStatus.METHOD_NOT_ALLOWED, bytes("Not found", "utf-8"), ExtensionRestResponse.TEXT_CONTENT_TYPE)
+        
+    @property
+    def routes(self) -> list[NamedRoute]:
+        return [NamedRoute(method=RestMethod.POST, path="/crud", unique_name="crud_post"),
+                NamedRoute(method=RestMethod.GET, path="/crud", unique_name="crud_get"),
+                NamedRoute(method=RestMethod.PUT, path="/crud", unique_name="crud_put"),
+                NamedRoute(method=RestMethod.DELETE, path="/crud", unique_name="crud_delete")]
```

Comment:
OK for now to just have the same path and use conditionals based on the method.  Eventually when #45 is implemented I think we should update this sample to use some params inline in the path.